openmmtools/integrators.py

```diff
@@ -1993,9 +1993,10 @@ def _add_integrator_steps(self):
         """
         # First step: Constrain positions and velocities and reset work accumulators and alchemical integrators
         self.beginIfBlock('step = 0')
-        self._add_update_alchemical_parameters_step() # sets alchemical parameters based on master lambda
         self.addConstrainPositions()
         self.addConstrainVelocities()
+        self._add_reset_protocol_work_step()
+        self._add_alchemical_reset_step() # set alchemical parameters too
```

Comment:
but this does that same, making the `reset_protocol_work_step` redundant once the cycle is complete

Reply:
i guess the only thing to make sure is that we extract the protocol work after every half cycle (forward or backward annealing) because the backward annealing continues to write to the protocol work variable; it might be advisable to zero the `protocol_work` once the `step` has reached `n_steps_eq + n_steps_neq` to avoid having to subtract out forward protocol work to get the backward protocol work?

Reply:
The trajectories may contain multiple cycles, so we would lose information that way.

I'm currently just extracting the work by 
```
forward_work = protocol_work[t1] - protocol_work[t0]
backward_work = protocol_work[t3] - protocol_work[t2]
```
etc., so it's pretty straightforward.

And yes, we can definitely remove redundancies and clean up the code here to make it faster and cleaner!